Description: Makefile fixes
 Set a custom SONAME and honour external CFLAGS/LDFLAGS
Author: Luca Bruno <lucab@debian.org>
Forwarded: no
---
--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,7 @@
 SRC_GENKAT = src/genkat.c
 OBJ = $(SRC:.c=.o)
 
-CFLAGS = -std=c89 -pthread -O3 -Wall -g
+CFLAGS += -std=c89 -pthread -O3 -Wall -g
 
 #OPT=TRUE
 ifeq ($(OPT), TRUE)
@@ -36,6 +36,7 @@
 ifeq ($(KERNEL_NAME), Linux)
 	LIB_EXT := so
 	LIB_CFLAGS := -shared -fPIC
+        SO_LDFLAGS := -Wl,-soname,libargon2.so.0
 endif
 ifeq ($(KERNEL_NAME), NetBSD)
 	LIB_EXT := so
@@ -63,7 +64,7 @@
 libs: $(LIB_SH) $(LIB_ST)
 
 $(RUN):	        $(SRC) $(SRC_RUN)
-		$(CC) $(CFLAGS) $^ -Isrc  -o $@
+		$(CC) $(CFLAGS) $(LDFLAGS) $^ -Isrc  -o $@
 
 $(BENCH):       $(SRC) $(SRC_BENCH)
 		$(CC) $(CFLAGS) $^ -Isrc  -o $@
@@ -72,7 +73,7 @@
 		$(CC) $(CFLAGS) $^ -Isrc  -o $@ -DGENKAT
 
 $(LIB_SH): 	$(SRC)
-		$(CC) $(CFLAGS) $(LIB_CFLAGS) $^ -Isrc -o $@
+		$(CC) $(CFLAGS) $(LIB_CFLAGS) $(LDFLAGS) $(SO_LDFLAGS) $^ -Isrc -o $@
 
 $(LIB_ST): 	$(OBJ)
 		ar rcs $@ $^
